### PR TITLE
[build] Fix non x64 linux builds

### PR DESCRIPTION
**Pull request by bobcao3**
Related issue = #4711

### CODE REVIEW

File: cmake/TaichiCore.cmake [hunk 1]
```diff
@@ -397,12 +397,14 @@ if (NOT WIN32)
         # Linux
         target_link_libraries(${CORE_LIBRARY_NAME} stdc++fs X11)
         target_link_libraries(${CORE_LIBRARY_NAME} -static-libgcc -static-libstdc++)
-        # Avoid glibc dependencies
-        if (TI_WITH_VULKAN)
-            target_link_libraries(${CORE_LIBRARY_NAME} -Wl,--wrap=log2f)
-        else()
-            # Enforce compatibility with manylinux2014
-            target_link_libraries(${CORE_LIBRARY_NAME} -Wl,--wrap=log2f -Wl,--wrap=exp2 -Wl,--wrap=log2 -Wl,--wrap=logf -Wl,--wrap=powf -Wl,--wrap=exp -Wl,--wrap=log -Wl,--wrap=pow)
+        if (TI_ARCH_X64)
```
Comment (qiao-bo) [inline cmake/TaichiCore.cmake:400]: Maybe also add `TI_ARCH_X64` as an option (default ON) on the top? so we don't have to specify at every non-aarch64 build. We have too many flags now..

**Review by qiao-bo — COMMENTED** [review]
Thanks for the quick fix.

File: cmake/TaichiCore.cmake (hunk 1 above)
Comment (bobcao3) [reply cmake/TaichiCore.cmake:400]: It is the same flag as the backend option. (This does complicate things, hmmm...) I should probably change this to the cmake host arch flags

File: cmake/TaichiCore.cmake [hunk 2]
```diff
@@ -397,12 +397,14 @@ if (NOT WIN32)
         # Linux
         target_link_libraries(${CORE_LIBRARY_NAME} stdc++fs X11)
         target_link_libraries(${CORE_LIBRARY_NAME} -static-libgcc -static-libstdc++)
-        # Avoid glibc dependencies
-        if (TI_WITH_VULKAN)
-            target_link_libraries(${CORE_LIBRARY_NAME} -Wl,--wrap=log2f)
-        else()
-            # Enforce compatibility with manylinux2014
-            target_link_libraries(${CORE_LIBRARY_NAME} -Wl,--wrap=log2f -Wl,--wrap=exp2 -Wl,--wrap=log2 -Wl,--wrap=logf -Wl,--wrap=powf -Wl,--wrap=exp -Wl,--wrap=log -Wl,--wrap=pow)
+        if (${CMAKE_HOST_SYSTEM_PROCESSOR} STREQUAL "x86_64")
```
Comment (qiao-bo) [inline cmake/TaichiCore.cmake:400]: This is fine as long as we don't distinguish between `host` vs `target` systems ;/

**Review by qiao-bo — APPROVED** [review]

**Pull request merged**